msgraph/identity_providers.go

```diff
@@ -0,0 +1,163 @@
+		BaseClient: NewClient(VersionBeta, tenantId),
+	}
+}
+
+// List returns a list of IdentityProviders.
+func (c *IdentityProvidersClient) List(ctx context.Context) (*[]IdentityProvider, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      "/identity/identityProviders",
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("IdentityProvidersClient.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("ioutil.ReadAll(): %v", err)
+	}
+	var data struct {
+		IdentityProviders []IdentityProvider `json:"value"`
+	}
+	if err := json.Unmarshal(respBody, &data); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	return &data.IdentityProviders, status, nil
+}
+
+// Create creates a new IdentityProvider.
+func (c *IdentityProvidersClient) Create(ctx context.Context, provider IdentityProvider) (*IdentityProvider, int, error) {
+	var status int
+	body, err := json.Marshal(provider)
+	if err != nil {
+		return nil, status, fmt.Errorf("json.Marshal(): %v", err)
+	}
+	resp, status, _, err := c.BaseClient.Post(ctx, PostHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusCreated},
+		Uri: Uri{
+			Entity:      "/identity/identityProviders",
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("IdentityProvidersClient.BaseClient.Post(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("ioutil.ReadAll(): %v", err)
+	}
+	var newProvider IdentityProvider
+	if err := json.Unmarshal(respBody, &newProvider); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	return &newProvider, status, nil
+}
+
+// Get retrieves an IdentityProvider.
+func (c *IdentityProvidersClient) Get(ctx context.Context, id string) (*IdentityProvider, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: Uri{
+			Entity:      fmt.Sprintf("/identity/identityProviders/%s", id),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, fmt.Errorf("IdentityProvidersClient.BaseClient.Get(): %v", err)
+	}
+	defer resp.Body.Close()
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, status, fmt.Errorf("ioutil.ReadAll(): %v", err)
+	}
+	var provider IdentityProvider
+	if err := json.Unmarshal(respBody, &provider); err != nil {
+		return nil, status, fmt.Errorf("json.Unmarshal(): %v", err)
+	}
+	return &provider, status, nil
+}
+
+// Update amends an existing IdentityProvider.
+func (c *IdentityProvidersClient) Update(ctx context.Context, provider IdentityProvider) (int, error) {
+	var status int
+	body, err := json.Marshal(provider)
```

Comment:
We should check the ID attribute to avoid a crash later on

```suggestion
	if provider.ID == nil {
		return status, errors.New("IdentityProvidersClient.Update(): cannot update identity provider with nil ID")
	}
	body, err := json.Marshal(provider)
```